packages/project/src/utils/__tests__/v3-utils.test.ts

```diff
@@ -0,0 +1,1762 @@
+                            [-102.0225, 41.1635],
+                            [-102.0261, 41.161],
+                            [-102.0256, 41.1624],
+                          ],
+                        ],
+                      },
+                      properties: {},
+                    },
+                  ],
+                },
+                cropYears: [
+                  {
+                    cropYear: 2000,
+                    crops: [
+                      {
+                        version: 2,
+                        cropName: 'soybean',
+                        type: 'annual crop',
+                        cropNumber: 1,
+                        classification: 'soybean',
+                        datePlanted: '04/20/2000',
+                        fertilizerEvents: [],
+                        harvestOrKillEvents: [
+                          {
+                            date: '09/14/2000',
+                            boundaryYield: 38,
+                            yieldNumeratorUnit: 'bu',
+                            yieldDenominatorUnit: 'ac',
+                          },
+                        ],
+                        irrigationEvents: [],
+                        limingEvents: [],
+                        organicMatterEvents: [],
+                        tillageEvents: [
+                          {
+                            classification: 'no tillage',
+                            type: 'no tillage',
+                            date: '04/20/2000',
+                          },
+                        ],
+                        burningEvents: [],
+                      },
+                    ],
+                  },
+                  {
+                    cropYear: 2001,
+                    crops: [
+                      {
+                        version: 2,
+                        cropName: 'corn',
+                        type: 'annual crop',
+                        cropNumber: 1,
+                        classification: 'corn',
+                        datePlanted: '04/20/2001',
+                        fertilizerEvents: [
+                          {
+                            date: '06/15/2001',
+                            productName: 'urea ammonium nitrate (30-00-00)',
+                            lbsOfN: 149.8,
+                            area: 6,
+                            quantityUnit: 'lbs/acre',
+                          },
+                        ],
+                        harvestOrKillEvents: [
+                          {
+                            date: '09/14/2001',
+                            boundaryYield: 134,
+                            yieldNumeratorUnit: 'bu',
+                            yieldDenominatorUnit: 'ac',
+                          },
+                        ],
+                        irrigationEvents: [
+                          {
+                            date: '04/25/2001',
+                            depth: undefined,
```

Comment:
Looking at it here I'd prefer that we omit the undefined properties rather than repeat them as undefined.  If the typing is causing problems then I'd just mark them all as nullable in `legacy-specification.ts`.

Sorry to change my tune on that, I know you asked me and I said to do it this way. :/